lib/fizzy/parser.cpp

```diff
@@ -534,6 +535,37 @@ Module parse(bytes_view input)
         throw parser_error("malformed binary: number of function and code entries must match");
 
     const auto total_func_count = module.imported_function_types.size() + module.funcsec.size();
+    const auto total_global_count =
+        module.imported_globals_mutability.size() + module.globalsec.size();
+
+    // Validate exports.
+    std::unordered_set<std::string_view> export_names;
+    for (const auto& export_ : module.exportsec)
+    {
+        switch (export_.kind)
+        {
+        case ExternalKind::Function:
+            if (export_.index >= total_func_count)
+                throw validation_error{"invalid index of an exported function"};
+            break;
+        case ExternalKind::Table:
+            if (export_.index != 0 || !module.has_table())
+                throw validation_error{"invalid index of an exported table"};
+            break;
+        case ExternalKind::Memory:
+            if (export_.index != 0 || !module.has_memory())
+                throw validation_error{"invalid index of an exported memory"};
+            break;
+        case ExternalKind::Global:
+            if (export_.index >= total_global_count)
+                throw validation_error{"invalid index of an exported global"};
+            break;
+        default:
+            assert(false);
+        }
+        if (const auto [_, inserted] = export_names.emplace(export_.name); !inserted)
```

Comment:
I so much dislike this syntax of C++17 🤷

Reply:
`if (!std::get<1>(export_names.emplace(export_.name))` isn't nicer.

But this entire thing is in a for loop, so why not just keep these as two statements (like the rest of codebase)?

Reply:
`export_names.emplace(export_.name).second` will work.

Reply:
changed it to @chfast's suggestion